Augmented Reality - Deathmatch/Assets/Scripts/GameManager.cs

```diff
@@ -6,48 +6,27 @@ public class GameManager : MonoBehaviour
 {
     private const int MaxLimitOfAvailableWarriors = 5;
 
-    private static GameManager _instance;
-
-    public static GameManager Instance
-    {
-        get
-        {
-            if (_instance == null)
-            {
-                Debug.LogError("Instance not specified");
-            }
-            return _instance;
-        }
-    }
-
     [SerializeField] private Warrior _warriorPrefab;
     [SerializeField] private GameObject _setupModeText;
     [SerializeField] private GameObject _destroyModeText;
     [SerializeField] private Text _numbersOfWarriorsOnSceneText;
     [SerializeField] private List<Warrior> _enemies;
     [SerializeField] private Toggle _toggleSetupMode;
     [SerializeField] private Toggle _toggleDestroyMode;
- 
-    private int _amountWarriors;
-
-    public List<Warrior> Enemies => _enemies;
-    public Toggle ToggleSetupMode => _toggleSetupMode;
-    public Toggle ToggleDestoryMode => _toggleDestroyMode;
 
+    private int _amountWarriors;
 
-    private void Awake()
+    public IEnumerable<Warrior> Enemies => _enemies;
+    public bool IsSetupState
     {
-        _instance = this;
+        get; private set;
     }
-
-    private void OnDestroy()
+    public bool IsDestroyState
```

Comment:
Use for that enum with 3 fields: Normal, Setup, Kill

Reply:
Enum with fields is not suitable here, because there can be two states at the same time.